hal/halx86/include/apic.h

```diff
@@ -1,253 +1,301 @@
-#define APIC_SIVR_FOCUS    (0x1 << 9)
-
-#define APIC_ESR_MASK      (0xFE << 0)    /* Error Mask */
-
-#define APIC_ICR0_VECTOR   (0xFF << 0)    /* Vector */
-#define APIC_ICR0_DM       (0x7 << 8)     /* Delivery Mode */
-#define APIC_ICR0_DESTM    (0x1 << 11)    /* Destination Mode */
-#define APIC_ICR0_DS       (0x1 << 12)    /* Delivery Status */
-#define APIC_ICR0_LEVEL    (0x1 << 14)    /* Level */
-#define APIC_ICR0_TM       (0x1 << 15)    /* Trigger Mode */
-#define APIC_ICR0_DESTS    (0x3 << 18)    /* Destination Shorthand */
-
-/* Delivery Modes */
-#define APIC_DM_FIXED	  (0x0 << 8)
-#define APIC_DM_LOWEST    (0x1 << 8)
-#define APIC_DM_SMI       (0x2 << 8)
-#define APIC_DM_REMRD     (0x3 << 8)
-#define APIC_DM_NMI       (0x4 << 8)
-#define APIC_DM_INIT      (0x5 << 8)
-#define APIC_DM_STARTUP   (0x6 << 8)
-#define APIC_DM_EXTINT	  (0x7 << 8)
-#define GET_APIC_DELIVERY_MODE(x)	(((x) >> 8) & 0x7)
-#define SET_APIC_DELIVERY_MODE(x,y)	(((x) & ~0x700) | ((y) << 8))
-
-/* Destination Shorthand values */
-#define APIC_ICR0_DESTS_FIELD          (0x0 << 0)
-#define APIC_ICR0_DESTS_SELF           (0x1 << 18)
-#define APIC_ICR0_DESTS_ALL            (0x2 << 18)
-#define APIC_ICR0_DESTS_ALL_BUT_SELF   (0x3 << 18)
-
-#define APIC_ICR0_LEVEL_DEASSERT (0x0 << 14) /* Deassert level */
-#define APIC_ICR0_LEVEL_ASSERT   (0x1 << 14) /* Assert level */
-
-#define GET_APIC_DEST_FIELD(x)   (((x) >> 24) & 0xFF)
-#define SET_APIC_DEST_FIELD(x)   (((x) & 0xFF) << 24)
-
-#define GET_APIC_TIMER_BASE(x)   (((x) >> 18) & 0x3)
-#define SET_APIC_TIMER_BASE(x)   ((x) << 18)
-#define APIC_TIMER_BASE_CLKIN    0x0
-#define APIC_TIMER_BASE_TMBASE   0x1
-#define APIC_TIMER_BASE_DIV      0x2
-
-#define APIC_LVT_VECTOR   		  (0xFF << 0)   /* Vector */
-#define APIC_LVT_DS       		  (0x1 << 12)   /* Delivery Status */
-#define APIC_LVT_REMOTE_IRR		  (0x1 << 14)	/* Remote IRR */
-#define APIC_LVT_LEVEL_TRIGGER		  (0x1 << 15)	/* Lvel Triggered */
-#define APIC_LVT_MASKED			  (0x1 << 16)   /* Mask */
-#define APIC_LVT_PERIODIC		  (0x1 << 17)   /* Timer Mode */
-
-#define APIC_LVT3_DM        (0x7 << 8)
-#define APIC_LVT3_IIPP      (0x1 << 13)
-#define APIC_LVT3_TM        (0x1 << 15)
-#define APIC_LVT3_MASKED    (0x1 << 16)
-#define APIC_LVT3_OS        (0x1 << 17)
-
-#define APIC_TDCR_TMBASE   (0x1 << 2)
-#define APIC_TDCR_MASK     0x0F
-#define APIC_TDCR_2        0x00
-#define APIC_TDCR_4        0x01
-#define APIC_TDCR_8        0x02
-#define APIC_TDCR_16       0x03
-#define APIC_TDCR_32       0x08
-#define APIC_TDCR_64       0x09
-#define APIC_TDCR_128      0x0A
-#define APIC_TDCR_1        0x0B
-
-#define APIC_TARGET_SELF         0x100
-#define APIC_TARGET_ALL          0x200
-#define APIC_TARGET_ALL_BUT_SELF 0x300
-
-#define APIC_INTEGRATED(version) (version & 0xF0)
-
-typedef enum {
-  amPIC = 0,    /* IMCR and PIC compatibility mode */
-  amVWIRE       /* Virtual Wire compatibility mode */
-} APIC_MODE;
-
-#ifdef CONFIG_SMP
-#define MAX_CPU   32
```

Comment:
This definition will probably be needed in future

Reply:
It's done via the MADT tables, so it being defined in the header file will end up redundant.